### PR TITLE
added support for the stddev, variance and geometric_mean aggregation fu...

### PR DESCRIPTION
**Pull request by mfelsche**
...nctions

### CODE REVIEW

File: docs/sql/aggregation.txt [hunk 1]
```diff
@@ -242,6 +242,94 @@ Example::
...
+
+``geometric mean`` is defined on all numeric types and on timestamp. It always
+returns double values. If a value is negative, all values were null or we got no
+value at all ``NULL`` is returned. If any of the aggregated values is ``0`` the result will be ``0.0``
+as well.
+
+.. note::
+
+    Due to java double precision arithmetic it is possible that any two executions
+    of the aggregation function on the same data produce slightly differing results.
```
Comment (mfussenegger) [inline docs/sql/aggregation.txt]: does this mean that the example below is flaky?

File: docs/sql/aggregation.txt [hunk 2]
```diff
@@ -242,6 +242,94 @@ Example::
...
+    |             2.2133638394 | Star System |
+    +--------------------------+-------------+
+    SELECT 3 rows in set (... sec)
+
+
+variance
+========
+
+The ``variance`` aggregation function computes the `Variance`_ of the set of non-null
+values in a column. It is a measure about how far a set of of numbers is spread.
```
Comment (mfussenegger) [inline docs/sql/aggregation.txt]: of of is repeated

File: sql/src/main/java/io/crate/operation/aggregation/statistics/moment/Kurtosis.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+package io.crate.operation.aggregation.statistics.moment;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * TODO
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/aggregation/statistics/moment/Kurtosis.java]: TODO?

File: sql/src/main/java/io/crate/operation/aggregation/statistics/moment/SkewNess.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+package io.crate.operation.aggregation.statistics.moment;
+
+import org.apache.commons.math3.util.FastMath;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * TODO
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/aggregation/statistics/moment/SkewNess.java]: TODO?

File: sql/src/main/java/io/crate/operation/aggregation/statistics/moment/StandardDeviation.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.aggregation.statistics.moment;
+
+import org.apache.commons.math3.util.FastMath;
+
+public class StandardDeviation extends Variance<StandardDeviation> {
+    @Override
+    public synchronized double result() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/aggregation/statistics/moment/StandardDeviation.java]: why is this synchronized?

File: sql/src/main/java/io/crate/operation/aggregation/statistics/moment/Variance.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+package io.crate.operation.aggregation.statistics.moment;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+
+import java.io.IOException;
+
+public class Variance<T extends Variance> implements Streamable {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/aggregation/statistics/moment/Variance.java]: remove generic

**Comment by mfussenegger** [comment]
ack

File: docs/sql/aggregation.txt (hunk 1 above)
Comment (mfelsche) [inline docs/sql/aggregation.txt]: nope, not that one

**Pull request merged**